### ISSUE TITLE
Incorrect correction of card names

### ISSUE THREAD

**Issue by RA-Kooi** [Cockatrice/Cockatrice#5037]
Card names are corrected with [this regex](https://github.com/Cockatrice/Cockatrice/blob/master/cockatrice/src/carddatabase.cpp#L271), but this regex only takes the case of ` // ` in account.  
In the Wixoss "plugin" we have various cards that have this naming scheme: `Character//Type`.
The `/` character should be removed anyway, because this is an illegal character on Linux systems,   
and I'm sure it's an issue on Windows as well. The cards are downloaded and displayed just fine  
however, but you can't override them using the custom pictures folder. It's not a huge issue if we  
need to hit Tomy's servers for some of the card art, but sometimes their servers are very slow and even  
drop connections.

I think the following regex should work: `([/*<>:"\\?\x00-\x08\x10-\x1f])`.

**Comment by ebbit1q** [comment]
you should read the next line https://github.com/Cockatrice/Cockatrice/blob/master/cockatrice/src/carddatabase.cpp#L272

the characters in the first regex are completely removed, the second regex is replaced with a space

why " // " is removed instead of replaced is only because it always did that and I didn't see a reason to change the behavior which would cause other people that already keep this strange rule in mind to need to change things.

you will be able to replace the names in Wixoss just fine, instead of `Character//Type` use `Character  Type`

**Comment by RA-Kooi** [comment]
Oh, I missed that in the second regex. Regexes aren't known for their readability :sweat_smile:

**Comment by ebbit1q** [comment]
this is true, neither would I call cockatrice's behavior strictly intuitive, but with these two cases for slashes aside most characters just get removed
